manifests/config.pp

```diff
@@ -4,24 +4,24 @@
 class qpid::config
 {
 
-  group { $qpid::group:
-    ensure => present,
-  }
-
   user { $qpid::user:
-    ensure => present,
+    ensure => $qpid::ensure,
     groups => $qpid::user_groups,
   }
 
+  group { $qpid::group:
+    ensure => $qpid::ensure,
+  }
```

Comment:
FYI: I'm not a fan of moving this around. If you need to move it, it implies a dependency. Relying on the order in the manifest is not the right way to achieve that.